src/overmapbuffer.cpp

```diff
@@ -532,42 +542,42 @@ std::vector<mongroup *> overmapbuffer::groups_at( const tripoint &p )
     return result;
 }
 
-std::array<std::array<scent_trace, 3>, 3> overmapbuffer::scents_near( const tripoint &origin )
+std::array<std::array<scent_trace, 3>, 3> overmapbuffer::scents_near( const tripoint_abs_omt
+        &origin )
 {
     std::array<std::array<scent_trace, 3>, 3> found_traces;
-    tripoint iter;
-    int x;
-    int y;
 
-    for( x = 0, iter.x = origin.x - 1; x <= 2 ; ++iter.x, ++x ) {
-        for( y = 0, iter.y = origin.y - 1; y <= 2; ++iter.y, ++y ) {
+    for( int x = -1; x <= 1 ; ++x ) {
```

Comment:
I think that is incorrect to use negative values for array indexes.

Reply:
Well spotted; this is wrong.

Reply:
I'll PR a fix.

Reply:
See #45291.